src/twisted/application/app.py

```diff
@@ -681,3 +690,17 @@ def startApplication(application, save):
         reactor.addSystemEventTrigger('after', 'shutdown', p.save, 'shutdown')
     reactor.addSystemEventTrigger('before', 'shutdown',
                                   service.IService(application).stopService)
+
+
+
+def exitWithSignal(sig):
```

Comment:
I don't think this needs to be public API.  We should be conservative about exporting new public API, preferring to go with private APIs at first and exposing them only deliberately.